src/plugins/telemetryTable/components/TelemetryFilterIndicator.vue

```diff
@@ -58,119 +57,105 @@
+                        return;
+                    }
+                });
+                return mixed;
             },
+            label() {
+                if (this.filtersMixed) {
+                    return FILTER_INDICATOR_LABEL_MIXED;
+                } else {
+                    return FILTER_INDICATOR_LABEL;
+                }
+            },
+            title() {
+                if (this.filtersMixed) {
+                    return FILTER_INDICATOR_TITLE_MIXED;
+                } else {
+                    return FILTER_INDICATOR_TITLE;
+                }
+            }
+        },
+        methods: {
             setFilterNames() {
                 let names = [];
+                let composition = this.openmct.composition.get(this.table.configuration.domainObject);
 
-                this.composition && this.composition.load().then((domainObjects) => {
+                composition && composition.load().then((domainObjects) => {
                     domainObjects.forEach(telemetryObject => {
                         let keyString= this.openmct.objects.makeKeyString(telemetryObject.identifier);
                         let filters = this.filteredTelemetry[keyString];
-                        this.telemetryKeyStrings.add(keyString);
+                        let metadataValues = this.openmct.telemetry.getMetadata(telemetryObject).values();
 
                         if (filters !== undefined) {
-                            let metadataValues = this.openmct.telemetry.getMetadata(telemetryObject).values();
-                            Object.keys(filters).forEach(key => {
-                                metadataValues.forEach(metadaum => {
-
-                                    if (key === metadaum.key) {
-                                        names.push(metadaum.name);
-                                    }
-                                });
-                            });
+                            this.collectFiltersName(_.omit(filters, [USE_GLOBAL]), metadataValues, names);
                         }
                     });
-                    this.filterNames = Array.from(new Set(names));
+                    this.filterNames = names.length === 0 ? names : Array.from(new Set(names));
                 });
             },
-            handleConfigurationChanges(configuration) {
-                if (!_.eq(this.filteredTelemetry, configuration.filters)) {
-                    this.updateFilters(configuration.filters || {});
-                }
-            },
-            checkFiltersForMixedValues() {
-                let valueToCompare = this.filteredTelemetry[Object.keys(this.filteredTelemetry)[0]];
-                let mixed = false;
-
-                Object.values(this.filteredTelemetry).forEach(value => {
-                    if (!_.isEqual(valueToCompare, value)) {
-                        mixed = true;
-                        return;
+            collectFiltersName(filters, metadataValues, names) {
```

Comment:
I wonder if it might be a little cleaner to have this method named something like "getFilterNamesFromMetadata()", which would return an array. It would reduce the number of arguments (because we're not passing in an array any more) and make its purpose a little clearer?

Reply:
@akhenry Yes, this part of the code could be cleaner. I renamed the method which now returns an array. Let me know what you think.